src/variables.jl

```diff
@@ -33,3 +33,98 @@ A [`AddVariableNotAllowed`](@ref) error is thrown if adding variables cannot be
 done in the current state of the model `model`.
 """
 add_variable(model::ModelLike) = throw(AddVariableNotAllowed())
+
+"""
+    add_constrained_variable(
+        model::ModelLike,
+        set::AbstractScalarSet
+    )::Tuple{MOI.VariableIndex,
+             MOI.ConstraintIndex{MOI.SingleVariable, typeof(set)}}
+
+Add to `model` a scalar variable constrained to belong to `set`, returning the
+index of the variable created and the index of the constraint constraining the
+variable to belong to `set`.
+
+By default, this function falls back to created a free variable with
+[`add_variable`](@ref) and then constrain it to belong to `set` with
+[`add_constraint`](@ref). Therefore, there is no need to implement this function
```

Comment:
The discussion below seems like it should be moved to the "implementing a solver interface" section in the manual.